protocol/contracts/amo/Ramos.sol

```diff
@@ -191,207 +264,304 @@ contract RAMOS is Ownable, Pausable {
+            if (quoteTokenAllowance < quoteTokenAmount) {
+                quoteToken.safeApprove(address(balancerVault), 0);
+                quoteToken.safeIncreaseAllowance(address(balancerVault), quoteTokenAmount);
+            }
+        }
 
         // join pool
-        uint256 bptAmountBefore = bptToken.balanceOf(address(this));
-        balancerVault.joinPool(balancerPoolId, address(this), address(this), request);
-        uint256 bptAmountAfter = bptToken.balanceOf(address(this));
         uint256 bptIn;
-        unchecked {
-            bptIn = bptAmountAfter - bptAmountBefore;
-        }
-        if (bptIn < minBptOut) {
-            revert AMOCommon.InsufficientAmountOutPostcall(minBptOut, bptIn);
+        {
+            uint256 bptAmountBefore = bptToken.balanceOf(address(this));
+            balancerVault.joinPool(balancerPoolId, address(this), address(this), request);
+            uint256 bptAmountAfter = bptToken.balanceOf(address(this));
+            unchecked {
+                bptIn = bptAmountAfter - bptAmountBefore;
+            }
         }
 
         // stake BPT
         bptToken.safeTransfer(address(amoStaking), bptIn);
         amoStaking.depositAndStake(bptIn);
+
+        emit LiquidityAdded(quoteTokenAmount, protocolTokenAmount, bptIn);
+    }
+
+    /**
+     * @notice Get the quote used to remove liquidity
+     * @dev Since this is not the view function, this should be called with `callStatic`
+     */
+    function proportionalRemoveLiquidityQuote(
+        uint256 bptAmount,
+        uint256 slippageBps
+    ) external override returns (
+        uint256 expectedProtocolTokenAmount,
+        uint256 expectedQuoteTokenAmount,
+        uint256 minProtocolTokenAmount,
+        uint256 minQuoteTokenAmount,
+        IBalancerVault.ExitPoolRequest memory requestData
+    ) {
+        return poolHelper.proportionalRemoveLiquidityQuote(bptAmount, slippageBps);
     }
 
     /**
-     * @notice Remove liquidity from balancer pool receiving both TEMPLE and stable tokens from balancer pool. 
-     * Treasury Price Floor is expected to be within bounds of multisig set range.
+     * @notice Remove liquidity from balancer pool receiving both `protocolToken` and `quoteToken` from balancer pool. 
+     * TPI is expected to be within bounds of multisig set range.
      * Withdraw and unwrap BPT tokens from Aura staking and send to balancer pool to receive both tokens.
      * @param request Request for use in balancer pool exit
      * @param bptIn Amount of BPT tokens to send into balancer pool
+     * @param to Address to which the `quoteToken` received from balancer pool are transferred
      */
     function removeLiquidity(
-        AMO__IBalancerVault.ExitPoolRequest memory request,
-        uint256 bptIn
-    ) external onlyOwner {
+        IBalancerVault.ExitPoolRequest memory request,
+        uint256 bptIn,
+        address to
+    ) external override onlyElevatedAccess {
```

Comment:
RAMOS has a number of guarded functions. perhaps we could use `payable` to save some gas.
compiler will ignore checks for if payment was provided.

Reply:
This is a lvl11 gas optimisation -- I'm not sure I follow this one sorry. What checks are ignored, how much gas andis it worth it for the extra confusion (for us, auditors, partners using RAMOS) since we'll only be accepting ERC20's?

Reply:
I think a lot of auditors are aware of this gas optimization. But if you think this creates confusion in the long run for others, then ignore.
Here's an article with some explanation 
https://dev.to/zaryab2000/does-adding-a-payable-keyword-in-solidity-actually-save-gas-24bm

Reply:
So just 24 gas saving? Seems a bit over the top imo

Reply:
Decided not to implement